core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1337,6 +1345,16 @@ public Clause maxClause() {
       return maxClause;
     }
 
+    /** return whether {@code Clause.GROUP_BY} is contained in {@code clauses} */
+    private boolean withGroupBy() {
+      for (Clause clause : clauses) {
+        if (clause == Clause.GROUP_BY) {
```

Comment:
This method is useless.

Reply:
I agree. `clauses.contains(Clause.GROUP_BY)` does the same job.